docs/draft/rest_api.md

```diff
@@ -0,0 +1,111 @@
+# First Draft for the REST API
+All calls and requests return data as JSON.
+## Data that needs to be provided by the API
+- current status of the transformer
+  ```
+  GET status/
+  ```
+  **returns:**
+  ```json
+  {
+    "state": "error|transformation|idle",
+    "current_transformations": [1,2,3]
+  }
+  ```
+  *state*: current status of the transformer
+  *current_transformations:* ids of the active or queued transformations
+- the result of the transformation: file/archive
+  ```
+  GET transformation/{id}
+  ```
+  **on success returns:**
+  ```json
+  {
+    "id": 1,
+    "result": "file"
+  }
+  ```
+- status of the current transformation
+  ```
+  GET transformation/{id}/status
+  ```
+  **on success returns:**
+  ```json
+  {
+    "id": 1,
+    "status": "inqueue|started|stopped|failed",
+    "progress":  "1-100"
```

Comment:
Just giving the progress value seems okay in combination with the logs. However i would still add a message field. this message field jst sends the current stage of the deployment. Like: Parsing, Generating Docker Images (For kubernetes)...

This also results in a question to the customer: Localisation of the application. If we just use english everywhere there is no need for localisation. however if we have to localize the frontend application we can't just send a text message.

This functionality could possibly be realized using the status field.